### DISCUSSION TITLE
想问一下关键点检测前后处理用resize代替仿射变换为什么结果可视化时位置不对。

### DISCUSSION THREAD

**Discussion by Eatzhy** [open-mmlab/mmpose#1195] [Q&A] [answered]

**Comment by ping-Huang** [comment]
This is the my test which compares with resize and warpaffine.
![](https://i.imgur.com/eRj0BUL.jpeg)
It seems like model can only accept non-deformable inputs.

**Comment by Eatzhy** [comment]
@ping-Huang old brother有微信吗，请教下，我测试的后处理用resize代替，关键点映射回原图有偏移。

**Comment by ping-Huang** [comment]
> @ping-Huang old brother有微信吗，请教下，我测试的后处理用resize代替，关键点映射回原图有偏移。

What postprocessing do you use?
Do you use the [TopDownAffine](https://github.com/open-mmlab/mmpose/blob/0376d51efdd93b3c8f3338211130753fed808bb9/mmpose/datasets/pipelines/top_down_transform.py#L230-L242) or [BottomUpResizeAlign](https://github.com/open-mmlab/mmpose/blob/0376d51efdd93b3c8f3338211130753fed808bb9/mmpose/datasets/pipelines/bottom_up_transform.py#L807-L812)?

**Comment by Eatzhy** [comment]
@ping-Huang TopDownAffine

**Comment by ping-Huang** [comment, marked as answer]
I've try some experiments below.
![](https://i.imgur.com/T5JCrSV.jpeg)
I'm not sure the left image is the bias which you means.

If you only want to use resize, you should get center, width and height [there](https://github.com/open-mmlab/mmpose/blob/75add9fc4fcad0a8eb77b29a62d44bd60732c534/mmpose/apis/inference.py#L99-L112).
Then, using 1.25 times of width and height to crop the image.
Finally, you can resize the crop image to model inputs.

I'm not sure why the input image should get the 1.25 times background padding.

Hope this experiment will be helpful for you.

**Reply by step404** [reply]
Hi, what's the meaning of 1.25?

**Reply by step404** [reply]
> Hi, what's the meaning of 1.25?

solved [here](https://mmpose.readthedocs.io/en/latest/faq.html)

**Comment by jin-s13** [comment]
Thanks @ping-Huang 
I will convert this issue to `discussion' @Eatzhy
